keras/src/saving/saving_lib.py

```diff
@@ -91,6 +112,37 @@ def save_model(model, filepath, weights_format="h5"):
         with open(filepath, "wb") as f:
             _save_model_to_fileobj(model, f, weights_format)
 
+def _save_and_upload_model_to_hf(model, hf_path, weights_format):
+    if hf_path.startswith("hf://"):
+        hf_path = hf_path[5:]
+    if hf_path.count("/") > 1:
+        raise ValueError(
+            "Invalid `hf_path` argument: expected `namespace/model_name` format. "
+            f"Received: hf_path={hf_path}"
+        )
+
+    api = huggingface_hub.HfApi(library_name="keras", library_version=keras_version)
+    repo_url = api.create_repo(hf_path, exist_ok=True)
+    repo_id = repo_url.repo_id
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        _save_model_to_folder(model, tmp_dir, weights_format)
+
+        model_card = _MODEL_CARD_TEMPLATE
+
+        if check_pydot():
+            plot_path = os.path.join(tmp_dir, "assets", "config.png")
+            plot_model(model, to_file=plot_path, show_layer_names=True, show_shapes=True, show_dtype=True)
+            model_card += "\n\n![](./assets/config.png)"
+
+        with open(os.path.join(tmp_dir, "README.md"), "w") as f:
+            f.write(model_card)
+
+        api.upload_folder(
+            repo_id=repo_id, folder_path=tmp_dir, commit_message="Save model using Keras."
+        )
+        print("Model saved to the Hugging Face Hub:", repo_url)
```

Comment:
Please use `io_utils.print_msg` instead of `print` for user-facing logs (so users can globally disable those if they're using a non-interactive setup where they need stdout to be clear)

Reply:
Will do! I added those prints when debugging locally. Addressed in https://github.com/keras-team/keras/pull/19854/commits/d436eefb96a4c1fd5a44b104011ee276504f6782